src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/SupplementaryDatasetService.java

```diff
@@ -28,7 +28,12 @@ public void addSupplementaryDatasetEntity(SupplementaryDatasetDTO supplementaryD
     CollectionExercise collectionExercise =
         collectionExerciseService.findCollectionExercise(
             supplementaryDatasetDTO.getSurveyId(), supplementaryDatasetDTO.getPeriodId());
-
+    if (collectionExercise == null) {
+      throw new CTPException(
+          CTPException.Fault.RESOURCE_NOT_FOUND,
+          "Failed to find collection exercise for supplementary dataset "
+              + "survey_id: {}, period_id: {}");
```

Comment:
The CTPException uses `String.format` which requires `%s` as the placeholders for interpolation rather than `{}`. We should include the values of the DTO under test in the assert of the message to ensure the `period_id` and `survey_id` are correctly included in the exception message.

Reply:
Have updated the code to use string.format, also made the tests more explicit by using the survey and period id strings to improve the testing.